### PR TITLE
Keep the crafting UI cursor in range after rebuilding the recipe list.

### LINKED ISSUE #18842: Crash in crafting UI when changing filter in batch mode
* Enter the crafting UI
* Hit [f], filter on a key that will match a few things (e.g "c:offal")
* choose an item other than the first one (e.g ."sweetbread")
* Hit [b] for batch mode
* Hit [f], filter on a different string that will match only one thing (e.g. "cornbread")

(Note that the UI still says sweetbread at this point)

* Hit [b] to exit batch mode

cataclysm coredumps:

```
/usr/include/c++/4.8/debug/vector:346:error: attempt to subscript container 
    with out-of-bounds index 1, but container only holds 1 elements.

Objects involved in the operation:
sequence "this" @ 0x0x7fff42bffaa0 {
  type = NSt7__debug6vectorIbSaIbEEE;
}
Aborted (core dumped)
```

stacktrace:

```
(gdb) bt
#0  0x00007fc040b9ec37 in __GI_raise (sig=sig@entry=6)
    at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007fc040ba2028 in __GI_abort () at abort.c:89
#2  0x00007fc0414f8fe5 in __gnu_debug::_Error_formatter::_M_error() const ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x0000000000603a67 in std::__debug::vector<bool, std::allocator<bool> >::operator[] (this=this@entry=0x7fff42bffaa0, __n=__n@entry=1)
    at /usr/include/c++/4.8/debug/vector:346
#4  0x0000000000600bc1 in select_crafting_recipe (
    batch_size=batch_size@entry=@0x7fff42c0086c: 0) at src/crafting_gui.cpp:375
#5  0x00000000005d966f in player::craft (this=0x1706a920)
    at src/crafting.cpp:89
#6  0x0000000000764009 in game::handle_action (this=this@entry=0x13951ee0)
    at src/game.cpp:3078
#7  0x00000000007677bc in game::do_turn (this=0x13951ee0) at src/game.cpp:1468
#8  0x00000000009840ed in main (argc=0, argv=0x7fff42c03a70)
    at src/main.cpp:468
```

### PR DESCRIPTION
**Pull request by mutability**
Fixes #18842

### CODE REVIEW

File: src/crafting_gui.cpp [hunk 1]
```diff
@@ -259,6 +259,13 @@ const recipe *select_crafting_recipe( int &batch_size )
                     return availability_cache[e];
                 } );
             }
+
+            // current/available have been rebuilt, make sure our cursor is still in range
+            if( current.empty() ) {
+                line = 0;
+            } else {
+                line = std::min( line, (int)current.size() - 1 );
```
Comment (illi-kun) [inline src/crafting_gui.cpp]: should be ```line = std::min( line, ( int )current.size() - 1 );``` to make astyle happy.

Comment (mutability) [inline src/crafting_gui.cpp]: Ok, cheers. my astyle is apparently "too old" so I can't check this before committing.

Comment (BorkBorkGoesTheCode) [inline src/crafting_gui.cpp]: Automatic linter:

http://dev.narc.ro/cataclysm/format.html

Comment (mutability) [inline src/crafting_gui.cpp]: I don't know what input that linter wants, but everything I've tried just gives me "syntax error"

Comment (BorkBorkGoesTheCode) [inline src/crafting_gui.cpp]: Are any of the lines broken at all? you need to delete the tabs. Does the file start and end with [ and ] ? 

Check every comma, quote, etc..

Comment (illi-kun) [inline src/crafting_gui.cpp]: @BorkBorkGoesTheCode linter is not applicable to cpp files. Keep in mind that you gives your tips for working with json files in wrong place.

**Pull request merged**